SwissDataScienceCenter/renku-ui
# feat(client): add WebSocket channel setup

This PR adds the client-side logic for WebSocket.

The WebSocket channel is set up and it gets re-created in case of recoverable errors (to be further expanded), and it listens to incoming messages.
Once a new message comes, it checks for a suitable handler and executes it.
Mind that we save minimal data to the store for the WebSocket channel itself, but the handlers can modify any part of the model.

This fully moves the logic to check for UI version updates to the WebSocket channel. It's also a good example of how we can send all the required data through the channel, although we probably want to avoid this behavior in most cases where we need to get a lot of data.

**How to test**
Open the network tab and verify that the WebSocket channel is created and information about the client version is sent regularly.

![Screenshot_20220829_173034](https://user-images.githubusercontent.com/43481553/187237957-a26aa1aa-5217-421c-a019-e3064535bfa2.png)

**Further development**
There will be a follow-up issue to finish and enable the (already partially coded) logic for finding session updates through the WebSocket channel.

/deploy renku-gateway=master #persist
fix #1338
